### DISCUSSION TITLE
Multiple default auth guards

### DISCUSSION THREAD

**Discussion by ol1s** [laravel/framework#40778] [Ideas]
Currently we can define multiple auth guards via route middleware params, whereby the guards are applied to the request in the order they are defined:

```php
Route::middleware('auth:guard1,guard2')->group(function () {
    Route::get('example', [ExampleController::class, 'action']);
});
```

It would be a nice feature to also be able to define multiple default guards in `config/auth.php`:

```php
//current
'defaults' => [
    'guard' => 'guard1',
],

//suggested
'defaults' => [
    'guard' => [
        'guard1',
        'guard2'
    ]
],
```

An example use case for needing multiple default guards would be a situation whereby an API can authenticate via either session or token in the request header.

**Edit:** in hindsight this is a bad idea becuase calling `Auth::guard()` with no args is a common use case and needs to resolve to a single default guard 😁

**Comment by ousid** [comment]
Why would you need multiple guards by default?

**Comment by shaedrich** [comment]
That kinda defeats the purpose of having a "default" wouldn't it?
